File: check.js
const readline = require("readline");
const solution = require("./main");
const fs = require("fs");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

rl.question("Enter 'RUN' or 'SUBMIT': ", function (button) {
  let file_name;
  if (button === "RUN") {
    file_name = "run_cases.txt";
  } else if (button === "SUBMIT") {
    file_name = "submit_cases.txt";
  } else {
    console.log("Invalid input");
    rl.close();
    return;
  }

  const results = [];
  let line_count = 0;

  const fileStream = fs.createReadStream(file_name);
  const rl = readline.createInterface({
    input: fileStream,
    crlfDelay: Infinity,
  });

  const resultStream = fs.createWriteStream("result.txt");

  rl.on("line", (line) => {
    line_count++;
    const [a, b, c] = line.split(" ").map(Number);
    const trueSum = c;
    const yourSum = solution(a, b);
    const result = yourSum === trueSum;
    results.push(result);
    if (!result) {
      console.log(`Case ${line_count} is false`);
      console.log(`Expected: ${trueSum}`);
      console.log(`Output: ${yourSum}`);
      resultStream.write(`Case ${line_count} is false\n`);
      resultStream.write(`Expected: ${trueSum}\n`);
      resultStream.write(`Output: ${yourSum}\n`);
    }
  });

  rl.on("close", () => {
    const all_results = results.every((r) => r);
    if (all_results) {
      console.log("Success");
      resultStream.write("Success\n");
    } else {
      console.log("Failure");
      resultStream.write("Failure\n");
    }
    resultStream.end();
  });
});

File: check.java
import java.io.*;
import java.util.*;

public class check {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter 'RUN' or 'SUBMIT': ");
        String button = "";
        button = scanner.nextLine().toUpperCase();
        String file_name = "";
        if(button.equals("RUN")){
            file_name = "run_cases.txt";
        } 
        if(button.equals("SUBMIT")){
            file_name = "submit_cases.txt";
        }  
        scanner.close();
        try (BufferedReader br = new BufferedReader(new FileReader(file_name));
                BufferedWriter bw = new BufferedWriter(new FileWriter("result.txt"))) {
            String line;
            List<Boolean> results = new ArrayList<>();
            int line_count = 0;
            while ((line = br.readLine()) != null) {
                ++line_count;
                String[] parts = line.split(" ");
                if (parts.length == 3) {
                    int a = Integer.parseInt(parts[0]);
                    int b = Integer.parseInt(parts[1]);
                    int c = Integer.parseInt(parts[2]);
                    int trueSum = c;
                    int yourSum = Main.solution(a, b);
                    boolean result = (yourSum == trueSum);
                    results.add(result);
                    if (!result) {
                        bw.write("Case " + line_count + " is false\n");
                        bw.write("Expected: " + trueSum + "\n");
                        bw.write("Output: " + yourSum + "\n");
                    }
                } else {
                    System.err.println("Error parsing line " + line_count);
                    System.exit(1);
                }
            }
            boolean all_results = true;
            for (boolean r : results) {
                if (!r) {
                    all_results = false;
                    break;
                }
            }
            if (all_results) {
                bw.write("Success\n");
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}